### PR TITLE
Skip the transfer page if the user has already selected a domain

### PR DESCRIPTION
**Pull request by rfaile313**
#### Changes proposed in this Pull Request

* Enhancement. Ref: Trello-Cobalt

- [x] If a user indicated that they want to transfer a domain name on the domain search page we don't need to show the add/transfer page because this repeats the same information the user already entered.

screenshot

#### Testing instructions

1. Go to My Site > Manage > Domains -> Add a Domain
2. Type domain that's already registered elsewhere, select "Yes, I own this domain"
screenshot
3. Clicking "Transfer to WordPress.com" is already fixed (forces precheck if the domain is transferrable)

### CODE REVIEW

**Comment by rfaile313** [comment]
Okay so I think I've set up a mapping check in [this commit](https://github.com/Automattic/wp-calypso/pull/32803/commits/a8ef624b3856121dbea783bd739d0579cb660ca3) by adding a check to `componentWillReceiveProps` in `client/my-sites/domains/map-domain/index.jsx` 

It definitely checks it but I'm not sure if this is good or what to do with the function `checkIfDomainIsTransferrable` ¯\_(ツ)_/¯

File: client/my-sites/domains/map-domain/index.jsx [hunk 1]
```diff
@@ -99,25 +100,25 @@ export class MapDomain extends Component {
...
-*/
 	checkSiteIsUpgradeable( props ) {
 		if ( props.selectedSite && ! props.isSiteUpgradeable ) {
 			page.redirect( '/domains/add/mapping' );
 		}
 	}
 
+	checkIfDomainIsTransferrable( props ) {
+		if ( props.lastDomainStatus === 'transferrable' ) {
+			//page.redirect( '/checkout' );
```
Comment (klimeryk) [inline client/my-sites/domains/map-domain/index.jsx]: Try something like `this.handleMapDomain( this.props.initialQuery );` :)

File: client/my-sites/domains/map-domain/index.jsx [hunk 2]
```diff
@@ -111,6 +113,12 @@ export class MapDomain extends Component {
...
```
Comment (klimeryk) [inline client/my-sites/domains/map-domain/index.jsx]: Try something like this.handleMapDomain( this.props.initialQuery );

File: client/my-sites/domains/map-domain/index.jsx [hunk 3]
```diff
@@ -111,6 +113,12 @@ export class MapDomain extends Component {
 		}
 	}
 
+	checkIfDomainIsTransferrable( props ) {
+		if ( props.lastDomainStatus === 'transferrable' ) {
```
Comment (klimeryk) [inline client/my-sites/domains/map-domain/index.jsx]: When you see yourself doing checks against raw strings like this, always try to check if there are already constants you could use. In this case, there are:

https://github.com/Automattic/wp-calypso/blob/fa95e755141615209f2fbb83f64d2775597eace5/client/lib/domains/constants.js#L58

(same for `mappable`)

File: client/my-sites/domains/map-domain/index.jsx [hunk 4]
```diff
@@ -111,6 +113,12 @@ export class MapDomain extends Component {
 		}
 	}
 
+	checkIfDomainIsTransferrable( props ) {
```
Comment (klimeryk) [inline client/my-sites/domains/map-domain/index.jsx]: In the mapping step, you should be checking/streamlining the `mappable` case, not the transferrable step.

File: client/my-sites/domains/map-domain/index.jsx (hunk 3 above)
Comment (rfaile313) [reply client/my-sites/domains/map-domain/index.jsx]: Perhaps I should see if I can change the transfer comparison string as well.

File: client/my-sites/domains/map-domain/index.jsx (hunk 4 above)
Comment (rfaile313) [reply client/my-sites/domains/map-domain/index.jsx]: Got it. I'll get after it!

**Comment by rfaile313** [comment]
Skipping confirmation step when a domain is specified in `Domains -> Add` is complete for both Transfer and Mapping.

Here's a video: 

screenshot

It isn't, however, ideal and still needs work:

Even though 'transferrable' and 'mappable' exist at `lib/domains/constants.js` they don't catch the comparison that `const forcePrecheck = this.props.lastDomainStatus === 'transferrable';` catches and I can't figure out why.

In /wp-calypso/client/my-sites/domains/map-domain/index.jsx I'm not sure how to bring in the `mappable` status in a logical way and we're already passing `transferrable` in the URL to compare it against.

tl;dr it might make more sense to just fix this and merge rather than let me continue trying to put a square peg into a round hole. I'm having trouble understanding the flow here.

**Comment by klimeryk** [comment]
Reworked the patch a bit. The transfer path is good to go. But the mapping one is a PITA. We need to do it in the controller, otherwise the component still flashes for a sec and that's crappy/lazy UX.
But it seems to be leading to an endless loop if you then try to remove the mapping from cart 🤦‍♂ Unless someone from @Automattic/cobalt has a better idea, I'd simply abandon this as not worth it.

**Comment by klimeryk** [comment]
Unfortunately, I missed how this impacts transfer-ins in some cases and has to be reverted :( https://github.com/Automattic/wp-calypso/pull/34411
